server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -372,18 +334,23 @@ private void mergeSharedOptions(FieldMapper mergeWith, List<String> conflicts) {
                 + "] to [" + mergeWith.contentType() + "]");
         }
 
-        MappedFieldType other = mergeWith.fieldType;
+        FieldType other = mergeWith.fieldType;
+        MappedFieldType otherm = mergeWith.mappedFieldType;
+        this.mappedFieldType.updateMeta(otherm.meta());
 
         boolean indexed =  fieldType.indexOptions() != IndexOptions.NONE;
         boolean mergeWithIndexed = other.indexOptions() != IndexOptions.NONE;
-        // TODO: should be validating if index options go "up" (but "down" is ok)
         if (indexed != mergeWithIndexed) {
             conflicts.add("mapper [" + name() + "] has different [index] values");
         }
+        // TODO: should be validating if index options go "up" (but "down" is ok)
+        if (fieldType.indexOptions() != other.indexOptions()) {
+            conflicts.add("mapper [" + name() + "] has different [index_options] values");
```

Comment:
new or existing error?

Reply:
Hm, I thought this was an existing error but it looks like it's a new one.  I'll add to the 'warnings in 7x' list.